lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -18,13 +18,12 @@ def lookup(path, vault_url = nil)
     # host is defined.
     raise Puppet::Error, "Unable to parse a hostname from #{vault_url}" unless uri.hostname
 
-    use_ssl = uri.scheme == 'https'
-    connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+    connection = Puppet.runtime[:http]
```

Comment:
#47 I just opened seems to show that the tests where working previously of this change.  I believe that the connector being changed, the behavior is not the same and the tests need some adjustment to test the right behavior.

I feel like `connection` was an established connection, and it is now just an HTTP client, so maybe some renaming can also help clarifying?